src/tls/test-connection.c

```diff
@@ -150,6 +157,33 @@ test_read_write (void)
 
   g_assert_cmpint (connection_read (c, WS), ==, RETRY);
 
+  /* test partial writes: send a large block */
+  memset (bigmsg, 42, sizeof (bigmsg));
+  g_assert_cmpint (send (client_fds[1], bigmsg, sizeof (bigmsg), 0), ==, sizeof (bigmsg));
+  g_assert_cmpint (connection_read (c, CLIENT), ==, SUCCESS);
+  g_assert_cmpint (c->buf_client.length, ==, sizeof (bigmsg));
+  /* write to ws should be partial due to our SO_SNDBUF from above */
+  g_assert_cmpint (connection_write (c, CLIENT), ==, PARTIAL);
+  g_assert_cmpint (c->buf_client.offset, >, 0);
+  g_assert_cmpint (c->buf_client.offset, <, sizeof (bigmsg));
+  /* flush the bigmsg to ws */
+  while (c->buf_client.offset > 0)
+    {
+      int len;
+
+      ConnectionResult r = connection_write (c, CLIENT);
+      if (r != SUCCESS && r != RETRY)
+        g_assert_cmpint (r, ==, PARTIAL);
+
+      for (;;)
+        {
+          len = recv (ws_fds[1], buffer, sizeof (buffer), 0);
```

Comment:
I could imagine an edge case where connection_write() fails to send data (some imaginary `EINTR` or whatever) and then this `recv()` will block here.  That's never going to happen in this test case, of course....

Reply:
The test does not have any child processes nor alarm()s, etc. Someone could send a SIGUSR1 or something at just the right time, but then again I'd rather have the test fail than introducing a while loop here that hides unexpected signals. I don't have a strong opinion here, other than that I prefer to be as simple and brittle as possible in tests. If you think we need an EINTR loop, I'll add one.